src/runtime/virtcontainers/clh.go

```diff
@@ -606,6 +656,10 @@ func (clh *cloudHypervisor) HotplugRemoveDevice(ctx context.Context, devInfo int
 	span, _ := katatrace.Trace(ctx, clh.Logger(), "HotplugRemoveDevice", clhTracingTags, map[string]string{"sandbox_id": clh.id})
 	defer span.End()
 
+	if clh.config.ConfidentialGuest {
+		return nil, errors.New("Device hotplug addition is not supported in confidential mode")
```

Comment:
nit -- s/addition/removal

Reply:
Oops, I will raise one PR fixing this, thanks!

Reply:
JFTR: https://github.com/kata-containers/kata-containers/pull/3776